plugins/woocommerce/includes/abstracts/abstract-wc-product.php

```diff
@@ -1762,7 +1763,17 @@ public function has_enough_stock( $quantity ) {
 	 * @return boolean
 	 */
 	public function has_attributes() {
```

Comment:
This is a method that is reading, not writing -- so the thinking from [this comment](https://github.com/woocommerce/woocommerce/pull/43739/files#r1457322310) wouldn't apply here.